### ISSUE TITLE
GVim - 'xapp and canberra' Modules failed

### ISSUE THREAD

**Issue by ViRodge** [vim/vim-appimage#50]
**os**: Arch
**kernel**: 6.2.8-arch1-1

Current package versions on system:
- **xapp**: 2.4.2-2
- **libcanberra**: 1:0.30+r2+gc0620e4-3
- **gdk-pixbuf2**: 2.42.10-2

----

**Gvim appimage does not want to open due to modules not loading.** 

```
Gtk-Message: 10:26:20.940: Failed to load module "xapp-gtk3-module"
Gtk-Message: 10:26:20.941: Failed to load module "canberra-gtk-module"

(gvim:76276): Gtk-WARNING **: 10:26:21.376: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /org/gtk/libgtk/icons/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /org/gtk/libgtk/icons/16x16/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Vim: Caught deadly signal ABRT
Vim: Finished.

Aborted (core dumped)
```
----
I did see there was `gdk-pixbuf-xlib` as a deprecated xlib integration package for GdkPixbuf in my Distro. I did installed that package as well and still gave the same error.
Currently `gdk-pixbuf-xlib` is on version 2.40.2-2

Also installed gvim locally from my Distros repo, and that one works fine. Thought it would change something with current packages but the appimage still errors out.

**Comment by chrisbra** [comment]
Are you using some kind of signs to display icons? Can you check if it works with `vim --clean`?

**Comment by ViRodge** [comment]
Downloaded the current Gvim appimage. I'm still getting the same problems and error. Sorry that I didn't state it crashes before. 
![image](https://user-images.githubusercontent.com/20025766/230732573-30098bae-756d-439a-8678-ac86667c6ace.png)

The Vim appimage works, however the GVim appimage is giving me the trouble. I've downloaded gvim through my distro and it works; GVim v9.0.1420-1.

----
I was curious if an older version worked. GVim-v9.0.1423.glibc2.29-x86_64.AppImage gives me the same error as the current version. Couldn't find the 1420 on the github so I went with GVim-v9.0.1413.glibc2.14-x86_64.AppImage. 

![image](https://user-images.githubusercontent.com/20025766/230733569-17542666-6f62-44f1-864b-200351d8c138.png)

I tried the gdk-pixbuf-query-loaders command it suggests but now 1413 just errors like the current Gvim-v9.0.1442 Appimage.
